### DISCUSSION TITLE
Flyway migration

### DISCUSSION THREAD

**Discussion by sroudebush-rr** [quarkusio/quarkus#30830] [Q&A]
I've got an app that builds an uber-jar and uses the generated main. This app has lots of rest endpoints, uses quartz for scheduled tasks, and runs flyway migrations on start. My problem is that I'd like to run the app in a mode such that it only runs the flyway migrations then immediately exits the app on success or failure of the migration. I'm wondering what the best approach would be to create such a mode like this for my app.

I've tried adding a custom `@QuarkusMain` class, that works but it seems difficult to disable other aspects of the app with this method.
I've also tried using the Flyway plugin but that requires it be bundled with my app to run.

[a comment by an automated account is not shown]

**Comment by gastaldi** [comment]
How about creating a Quarkus app having only the main class and the flyway extension?

**Reply by geoand** [reply]
I think @iocanel introduced a way to do this

**Reply by sroudebush-rr** [reply]
Ah yeah, I see where that change was introduced: https://github.com/quarkusio/quarkus/commit/923dcae38f9335b8d3e51b7b53d3eb5309c05828
I'll check that out

**Reply by jonathanrapid** [reply]
how is the startup sequence ordered for quarkus? do we know if setting the flag for exiting after the flyway migration gives us sufficient guarantees that other beans haven't yet started up and "doing business stuff"?

**Reply by sraturi** [reply]
@gastaldi  any information on this? https://github.com/quarkusio/quarkus/discussions/30830#discussioncomment-4856135

**Reply by gastaldi** [reply]
Yes, it should exit cleanly after all initialization tasks are completed

**Reply by sraturi** [reply]
what version is this used in? I tried `3.0.0-Alpha4` but ran into [this comment](https://github.com/quarkusio/quarkus/pull/29026/files#r1039485648) Also I don't see this field being referenced anywhere else i.e `FlywayContainer` in version `3.0.0-Alpha4`

**Reply by gastaldi** [reply]
Have you tried running your app with `-Dquarkus.flyway.run-and-exit=true`? See https://quarkus.io/version/main/guides/flyway#quarkus-flyway_quarkus.flyway.run-and-exit

**Reply by sraturi** [reply]
yeah tried setting `quarkus.flyway.run-and-exit=true` in the application properties but the application just continues running.

**Reply by iocanel** [reply]
`QUARKUS_INIT_AND_EXIT` should allow the application to exit right after initialization (e.g. flyway, liquibase etc take place).

**Reply by iocanel** [reply]
Created: https://github.com/quarkusio/quarkus/pull/31259 to remove the unused property `run-and-exit`.

**Reply by sraturi** [reply]
@iocanel  thanks! that seems to work.  Although I don't see the process exit code..? 
Also, this seems to be available around `v3.0.0-Alpha4` ? any idea if something similar exists for  ~~ `v2.15.2`
